lib/hls/hls_parser.js

```diff
@@ -3219,12 +3225,20 @@ shaka.hls.HlsParser = class {
     let previousReference = null;
     let lastDiscontinuityStartTime = firstStartTime;
 
+    const bitrates = [];
+
     for (let i = 0; i < hlsSegments.length; i++) {
       const item = hlsSegments[i];
       const startTime =
           (i == 0) ? firstStartTime : previousReference.endTime;
       position = mediaSequenceNumber + skippedSegments + i;
 
+      const bitrate = shaka.hls.Utils.getFirstTagWithNameAsNumber(
+          item.tags, 'EXT-X-BITRATE');
+      if (bitrate) {
+        bitrates.push(bitrate);
```

Comment:
This isn't enough info to compute an average.  You also need the duration of the segment.  What you want is not AVG(bitrates), but SUM(bitrate * duration) / SUM(duration)

Reply:
Fixed!